ldms/python/ldmsd/ldmsd_communicator.py

```diff
@@ -1639,26 +1639,26 @@ def store_time_stats(self, name=None, reset = False):
         except Exception as e:
             return errno.ENOTCONN, str(e)
 
-    def plugn_load(self, name, instance=None):
+    def plugn_load(self, name, plugin=None):
         """
         Load a plugin instance.
 
         Parameters:
-        name  - The plugin name
-        inst  - The configuration instance name. If None, 'name' is used
+        name  - The instance name
+        plugin- The plugin name. If None, 'name' is used
 
         Returns:
         A tuple of status, data
         - status is an errno from the errno module
         - data is an error message if status != 0 or None
         """
-        if not instance:
-            instance = name
+        if not plugin:
```

Comment:
I think we should drop this check. It is redundant, since ldmsd is has the logic to handle when plugin is not provided. In fact having this duplicated logic here makes it difficult to test if the server code is doing what it should, because ldmsd_controller won't let us send without "plugin" set.

What I did was remove this check, and only optionally add include the "LDMSD_Req_Attr" for plugin when plugin is not None.